pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1493,14 +1493,24 @@ func (c *PodAffinityChecker) satisfiesPodsAffinityAntiAffinity(pod *v1.Pod,
 			}
 		}
 
+		allTermMatches := true
 		// Check all anti-affinity terms.
 		for _, term := range GetPodAntiAffinityTerms(affinity.PodAntiAffinity) {
 			termMatches, _, err := c.anyPodMatchesPodAffinityTerm(pod, filteredPods, nodeInfo, &term)
-			if err != nil || termMatches {
+			if err != nil {
 				glog.V(10).Infof("Cannot schedule pod %+v onto node %v, because of PodAntiAffinityTerm %v, err: %v",
 					podName(pod), node.Name, term, err)
 				return ErrPodAntiAffinityRulesNotMatch, nil
 			}
+			if !termMatches {
```

Comment:
This change won't fix the issue. For example, if there are two pods, P1 and P2 and there are two anti-affinity terms, T1 and T2, where P1 only matches T1 and P2 only matches T2, `allTermMatches` will be true while there is no pod that matches both T1 and T2.
I just sent out #62715 to fix this issue. Sorry that I hadn't mentioned earlier that I would be working on this.

Reply:
Great, thanks for your explanation, I will close it.